src/libAtomVM/nifs.c

```diff
@@ -1685,7 +1685,7 @@ static term nif_erlang_binary_to_list_1(Context *ctx, int argc, term argv[])
         RAISE_ERROR(OUT_OF_MEMORY_ATOM);
     }
 
-    const char *bin_data = term_binary_data(argv[0]);
+    const uint8_t *bin_data = (const uint8_t *) term_binary_data(argv[0]);
```

Comment:
This doesn't seem a workaround but an incomplete fix to me. Indeed, the complete fix would be to also fix the result of `term_binary_data` which should be `const void *` in which case we would still need the cast.

`binary_to_list/1` really is about `uint8_t` bytes with unsigned values from 0 to 255 instead of signed `char` between -128 and 127.

Reply:
I agree with @pguyot .  If anything, let's put a TODO comment on the term_binary_data function to address the type in a future PR.  Otherwise, what we are doing with signed chars is wrong (and has tripped me up in the past)